### PR TITLE
Hackweek: Added Makefile entry to fetch schema; eliminate type errors

### PR DESCRIPTION
**Pull request by dbennett-sentry**

### CODE REVIEW

File: snuba/datasets/events_processor_base.py [hunk 1]
```diff
@@ -137,11 +140,13 @@ def process_message(
         Returns `None` if the event is too old to be written.
         """
         action_type = None
+        processed: Optional[MutableMapping[str, Any]] = None
 
         if isinstance(message, dict):
             # deprecated unwrapped event message == insert
             action_type = ProcessorAction.INSERT
             try:
+
```
Comment (fpacifici) [inline snuba/datasets/events_processor_base.py:149]: intentional ?

File: snuba/datasets/events_processor_base.py [hunk 2]
```diff
@@ -228,7 +234,7 @@ def process_insert(
         self.extract_common(processed, event, metadata)
         self.extract_custom(processed, event, metadata)
 
-        sdk = data.get("sdk", None) or {}
+        sdk = data.get("sdk", None) or {}  # type: ignore
```
Comment (fpacifici) [inline snuba/datasets/events_processor_base.py:237]: What was the issue here ?

**Review by fpacifici — APPROVED** [review]
cool

File: snuba/datasets/events_processor_base.py [hunk 3]
```diff
@@ -273,15 +283,15 @@ def extract_common(
 
         # Properties we get from the "data" dict, which is the actual event body.
 
-        received = _collapse_uint32(int(data["received"]))
+        received = _collapse_uint32(data["received"])
```
Comment (dbennett-sentry) [inline snuba/datasets/events_processor_base.py:286]: The extra `int` here is probably a bug as `int(None)` fails, `_collapse_uint32(None)` returns `None`, and the subsequent code checks for `None`

File: snuba/datasets/events_processor_base.py (hunk 1 above)
Comment (dbennett-sentry) [reply snuba/datasets/events_processor_base.py:158]: Nope.

File: snuba/datasets/events_processor_base.py (hunk 2 above)
Comment (dbennett-sentry) [reply snuba/datasets/events_processor_base.py:246]: "sdk" does not exists on the data object.

File: snuba/datasets/events_processor_base.py [hunk 4]
```diff
@@ -28,11 +29,26 @@
...
+    from typing import TypedDict  # pylint: disable=no-name-in-module
+else:
+    from typing_extensions import TypedDict
 
 logger = logging.getLogger(__name__)
 
+EventData = JSONSchema["schema/event.schema.json"]
+
+
+class Event(TypedDict):
```
Comment (fpacifici) [inline snuba/datasets/events_processor_base.py:43]: I think this class is duplicate of InsertEvent below. We should have only one.

**Pull request merged**